common/src/service/ExceptionHandlingHelper.cs

```diff
@@ -17,97 +17,88 @@ namespace Microsoft.Azure.Devices
 {
     internal class ExceptionHandlingHelper
     {
-        public static IDictionary<HttpStatusCode, Func<HttpResponseMessage, Task<Exception>>> GetDefaultErrorMapping()
+        private static readonly IReadOnlyDictionary<HttpStatusCode, Func<HttpResponseMessage, Task<Exception>>> s_mappings =
```

Comment:
Instead of creating this dictionary for every call, make it static/readonly/readonlydictionary and return the same instance.